x-pack/plugins/entity_manager/server/lib/v2/constants.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// Definitions index
+
+export const DEFINITIONS_ALIAS = '.kibana-entities-definitions';
+export const TEMPLATE_VERSION = 1;
+
+// Privileges
+
+export const CREATE_ENTITY_TYPE_DEFINITION_PRIVILEGE = 'create_entity_type_definition';
+export const CREATE_ENTITY_SOURCE_DEFINITION_PRIVILEGE = 'create_entity_source_definition';
+export const READ_ENTITY_TYPE_DEFINITION_PRIVILEGE = 'read_entity_type_definition';
+export const READ_ENTITY_SOURCE_DEFINITION_PRIVILEGE = 'read_entity_source_definition';
+export const READ_ENTITIES_PRIVILEGE = 'read_entities';
```

Comment:
these privileges are applied at the rest api layer, what about client layer ? do we have to implement a similar mechanism there or do we expect consuming plugins to verify privileges before hand ?

Reply:
Great question. I don't believe the entity client itself can/should manage these kinds of authorization checks so I would push that to the integrating apps to handle for their APIs that then use the client.
In order to do that, we'll also need to expose these privileges so they can be re-used so I'll do that.

Reply:
In https://github.com/elastic/kibana/pull/201470/commits/abc12bdcb3ded246fae5eb145723c7bb5ccbe7a7 I'm exporting them from the plugin index, I don't think we need to attach it to the start contract of the plugin since it's not dependent on the life cycle so a `requiredBundle` config should be enough.

Reply:
Sounds good. thinking a consuming plugin could want to abstract entity manager and create its own set of privileges, so that approach makes that possible while also allowing reuse of our own privileges